src/js/legend.js

```diff
@@ -0,0 +1,228 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.defaults("gpii.chartAuthoring.pieChart.legend", {
+        gradeNames: ["gpii.d3ViewComponent", "autoInit"],
```

Comment:
While I'm not absolutely sure about this, I believe the d3.legend referenced here is actually a user-created plugin, rather than a core of D3. See https://github.com/jieter/d3-legend - also no mention of it in the D3 API reference at https://github.com/mbostock/d3/wiki/API-Reference

I think the main rationale for drawing the legend with D3 is to use the same paradigm to binding data to DOM elements and managing updates from the model as is used for the pie. At least to my mind, the legend is another form of data visualization, so appropriately bounded by how we've chosen to use D3.